pkg/reconciler/reconciler.go

```diff
@@ -567,8 +571,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
 	u.UpdateStatus(updater.EnsureCondition(conditions.Initialized(corev1.ConditionTrue, "", "")))
 
 	if obj.GetDeletionTimestamp() != nil {
-		err := r.handleDeletion(ctx, actionClient, obj, log)
-		return ctrl.Result{}, err
+		if err := r.handleDeletion(ctx, actionClient, obj, log); err != nil {
+			return ctrl.Result{}, err
+		}
+		shouldUpdate = false
```

Comment:
This is slightly different from the case described in the PR description, right?
I'm guessing the problem here was that on deletions both the `uninstallUpdater` and the regular `u` updater would fire, and the latter was both useless and would cause errors since the resource would typically be already gone after deleting the finalizer?
If I got this right, please mention this in the PR description.